Wrappers/Python/cil/processors/CofR_xcorrelation.py

```diff
@@ -70,54 +78,76 @@ def check_input(self, data):
 
             if self.slice_index < 0 or self.slice_index >= data.get_dimension_size('vertical'):
                 raise ValueError('slice_index is out of range. Must be in range 0-{0}. Got {1}'.format(data.get_dimension_size('vertical'), self.slice_index))
-
-        if self.projection_index >= data.geometry.config.angles.num_positions:
-                raise ValueError('projection_index is out of range. Must be less than {0}. Got {1}'.format(data.geometry.config.angles.num_positions, self.projection_index))
+              
+        # check if projection_index is either a tuple or list of length 2       
+        try:
+            len_check = False if len(self.projection_index) != 2 else True
+            index = list(self.projection_index)
+        # if projection_index does not have a length, put the object in a list
+        except:
+            len_check = True
+            index = [self.projection_index]
+        
+        if not len_check:
+            raise ValueError('Expected projection_index to be an int or list/tuple of 2 ints, got {0}'.format(self.projection_index))
+        
+        for angle in index:
+            # check if all the indices are int
+            if not isinstance(angle, int):
+                raise ValueError('Expected projection_index to be an int, got {0}'.format(type(angle)))
+            
+            # check if all the indices are in range 0 to the number of angles
+            if angle< 0  or angle>=data.geometry.config.angles.num_positions:
+                raise ValueError('projection_index is out of range. Must be between 0 and {0}. Got {1}'.format(data.geometry.config.angles.num_positions-1, self.projection_index))
+
+        angles_deg = data.geometry.config.angles.angle_data.copy()
+        if data.geometry.config.angles.angle_unit == "radian":
+                angles_deg *= 180/np.pi 
+
+        # if there is only 1 index specified, find the angle in the list closest to 180 degrees away from the index 
+        if len(index) == 1:
+            new_index = CofR_xcorrelation._return_180_index(angles_deg, self.projection_index)
+            index.append(new_index)
+        
+        # check the two angles are 180 degrees apart within the specified tolerance
+        ang_diff = (angles_deg[index[0]] - angles_deg[index[1]]) % 360
+        if abs(ang_diff-180) > self.ang_tol:
```

Comment:
This error message might be confusing if you provided the indices for the corrector. 
If they provide two indices but the difference between the angles is not within tolerance what should happen? Maybe raise an error that says "'Method requires projections 180+/-{0} degrees apart. You provided projection angle {1} and angle {2} that are not within this tolerance.\
                            \nPick different projections or increase the angular tolerance `ang_tol`.'.format(self.ang_tol, data.geometry.angles[index[0]], data.geometry.angles[index[1]]))

Reply:
Good point, I've updated this now